relay/config.go

```diff
@@ -1,14 +1,61 @@
 package relay
 
-import core "github.com/Layr-Labs/eigenda/core/v2"
+import (
+	"fmt"
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/common/aws"
+	core "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/spf13/viper"
+	"os"
+	"strings"
+)
+
+const (
+	// relayEnvPrefix is the prefix for all environment variables used by the relay.
+	relayEnvPrefix = "RELAY"
+)
 
 // Config is the configuration for the relay Server.
+//
+// Environment variables are mapped into this struct by taking the name of the field in this struct,
+// converting to upper case, and prepending "RELAY_". For example, "BlobCacheSize" can be set using the
+// environment variable "RELAY_BLOBCACHESIZE".
+//
+// For nested structs, add the name of the struct variable before the field name, separated by an underscore.
+// For example, "Log.Format" can be set using the environment variable "RELAY_LOG_FORMAT".
+//
+// Slice values can be set using a comma-separated list. For example, "RelayIDs" can be set using the environment
+// variable "RELAY_RELAYIDS='1,2,3,4'".
+//
+// It is also possible to set the configuration using a configuration file. The path to the configuration file should
+// be passed as the first argument to the relay binary, e.g. "bin/relay config.yaml". The structure of the config
+// file should mirror the structure of this struct, with keys in the config file matching the field names
+// of this struct.
 type Config struct {
 
+	// Log is the configuration for the logger. Default is common.DefaultLoggerConfig().
+	Log common.LoggerConfig
+
+	// Configuration for the AWS client. Default is aws.DefaultClientConfig().
+	AWS aws.ClientConfig
+
 	// RelayIDs contains the IDs of the relays that this server is willing to serve data for. If empty, the server will
 	// serve data for any shard it can.
 	RelayIDs []core.RelayKey
 
+	// GRPCPort is the port that the relay server listens on. Default is 50051. // TODO what is a good port?
+	GRPCPort int
+
+	// BucketName is the name of the S3 bucket that stores blobs. Default is "relay".
```

Comment:
Are chunks and blobs in different buckets? If so, might be better to make the names more specific

Reply:
No, they are in the same bucket. We discussed this as a group when you were on PTO, and came to the conclusion that it was simpler to use a single bucket. We use [this code](https://github.com/Layr-Labs/eigenda/blob/master/common/aws/s3/scoped_keys.go) to ensure we don't have collisions between different data types.